helm.el

```diff
@@ -2449,7 +2449,31 @@ frame configuration as per `helm-save-configuration-functions'."
                (let ((frame (if (minibufferp helm-current-buffer)
                                 (selected-frame)
                               (last-nonminibuffer-frame))))
-                 (select-frame-set-input-focus frame))))))
+                 ;; This is a workaround, because the i3 window
+                 ;; manager developers are refusing to fix their
+                 ;; broken timestamp and event handling.
+                 ;;
+                 ;; We basically just disable the part of
+                 ;; select-frame-set-input-focus that would call
+                 ;; XSetInputFocus in Xlib (x-focus-frame), that
+                 ;; resets a timestamp in the xserver which the i3
+                 ;; developers fail to notice.
+                 ;;
+                 ;; Since they don't know about the new timestamp,
+                 ;; their keyboard handling can break after a helm
+                 ;; user quits emacs, as reported in #1641.
+                 ;;
+                 ;; Fortunately for us, we really don't need this
+                 ;; XSetInputFocus call, since we already have focus
+                 ;; for Emacs, the user is just using helm!  We call
+                 ;; select-frame-set-input-focus for the other
+                 ;; side-effects, not for x-focus-frame.
+                 (let ((saved-x-focus-frame (symbol-function 'x-focus-frame)))
+                   (unwind-protect
+                       (progn
+                         (fset 'x-focus-frame #'(lambda (_)))
+                         (select-frame-set-input-focus frame))
+                     (fset 'x-focus-frame saved-x-focus-frame))))))))
```

Comment:
IIUC, you want to disable `x-focus-frame` when calling `select-frame-set-input-focus` ? Right?
If so you can use this instead:

```elisp
(cl-letf (((symbol-function 'x-focus-frame) #'ignore))
  (Do_what_you_have_to_do_here))
```